client.go

```diff
@@ -108,6 +108,11 @@ func (c *Client) Delete(ctx context.Context, path string, query Query, recv inte
 // has an empty body. The recv argument should be a pointer to a type which
 // matches the format of the response, or nil if no response is expected.
 func (c *Client) Do(ctx context.Context, method string, path string, query Query, send interface{}, recv interface{}) (err error) {
+	_, err = c.doh(ctx, method, path, query, send, recv)
+	return
+}
+
+func (c *Client) doh(ctx context.Context, method string, path string, query Query, send interface{}, recv interface{}) (header http.Header, err error) {
```

Comment:
Should we just modify the signature of `do` to always return the header? Or maybe always decode `X-Consul-Index` and return that value instead of the full header?

Reply:
There are some other potentially valuable headers that they return, but I'm not sure what the best way to handle them is.

I really wish they had integrated this into the response struct since it feels hacky to be parsing and returning headers in the client library.

I'll defer to you for the best way to do it.  The simplest is a breaking change which just returns the headers or a struct with the values we care about.  I don't think there's a best way to handle this.

Reply:
How about adding a struct type for all info that Consul returns in the HTTP header and always parse them?
```go
type info struct {
    index int
    ...
}
```
```go
func (c *Client) do(...) (info, error) {
   ...
}
```
Anywhere the info are not used we can just do `_, err = client.do(...)`.

Reply:
I'm on board with this.

Grepping the consul source, I see these:

```
X-Consul-Index
X-Consul-LastContact
X-Consul-Translate-Addresses
X-Consul-KnownLeader
```

Not including the token header.

Reply:
Let's add this 👍